scripts/simulator.ts

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env node -r esm -r ts-node/register
+// noinspection ES6PreferShortImport,JSVoidFunctionReturnValueUsed
+
+import { strict as assert } from "assert";
+
+import chalk from "chalk";
+import { differenceInSeconds, parseISO, differenceInMinutes } from "date-fns";
+
+import {
+  ensureBotUsers as actualEnsureBotUsers,
+  removeBotUsers,
+  removeBotReactions,
+  removeBotChatMessages,
+} from "./lib/bot";
+import { getReactionsRef, getChatlinesRef } from "./lib/collections";
+import { getVenueRef } from "./lib/documents";
+import { withErrorReporter } from "./lib/log";
+import { run } from "./lib/runner";
+import {
+  CollectionReference,
+  DocumentReference,
+  RunContext,
+  SimConfig,
+} from "./lib/types";
+
+import { simChat } from "./simulation/chat";
+import { simExperience } from "./simulation/experience";
+import { simSeat } from "./simulation/seat";
+
+export type MainResult = {
+  chatsRef?: CollectionReference;
+  reactionsRef?: CollectionReference;
+  userRefs: DocumentReference[];
+};
+
+export type CleanupOptions = RunContext<SimConfig> & { result: MainResult };
```

Comment:
`run` function will call the `cleanup` one with all the context `main` got plus the `result` of `main`